nucypher/network/nodes.py

```diff
@@ -239,17 +237,17 @@ class NodeSprout(PartiallyKwargifiedBytes):
 
     def __init__(self, node_metadata):
         super().__init__(node_metadata)
-        self.checksum_address = to_checksum_address(node_metadata['public_address'][0])
+        self.checksum_address = to_checksum_address(self.public_address)
```

Comment:
:tada: - Looking excellent.

Reply:
...and much more optimal.  Saves a keccak and other smaller cpu-bound operations compared to the previous method.

Reply:
Really nice

Reply:
With other changes too, readability is certainly improved!